File: algolab/12주차/양팔저울.cpp
#include <iostream>
#include <vector>
#include <cmath>
#include <algorithm>

using namespace std;

int calculate_weights(const vector<int>& pebbles) {
    int left = pebbles[0];
    int right = pebbles[1];

    for (size_t i = 2; i < pebbles.size(); ++i) {
        if (left <= right) {
            left += pebbles[i];
        } else {
            right += pebbles[i];
        }
    }

    int diff = abs(left - right);

    vector<int> weights = {100, 50, 20, 10, 5, 2, 1}; // 리스트 초기화
    int count = 0;
    for (int weight : weights) {
        if (diff == 0) break;
        count += diff / weight;
        diff %= weight;
    }

    return count;
}

int main() {
    int t;
    cin >> t;

    vector<int> results;
    for (int test = 0; test < t; ++test) {
        int n;
        cin >> n;

        vector<int> pebbles(n);
        for (int i = 0; i < n; ++i) {
            cin >> pebbles[i];
        }

        results.push_back(calculate_weights(pebbles));
    }

    for (int result : results) {
        cout << result << endl;
    }

    return 0;
}

File: algolab/12주차/양팔저울.py
def calculate_weights(pebbles):
    left = pebbles[0]  # 첫 번째 자갈을 왼쪽에
    right = pebbles[1]  # 두 번째 자갈을 오른쪽에

    # 3번 자갈부터 배치
    for i in range(2, len(pebbles)):
        if left == right:
            left += pebbles[i]
        elif left < right:
            left += pebbles[i]
        else:
            right += pebbles[i]

    # 최종적으로 양쪽 무게 차이를 구함
    diff = abs(left - right)

    # 필요한 추가 무게추 개수 계산
    weights = [100, 50, 20, 10, 5, 2, 1]
    count = 0
    for weight in weights:
        if diff == 0:
            break
        count += diff // weight
        diff %= weight

    return count

# 입력 처리
t = int(input())
results = []
for _ in range(t):
    n = int(input())
    pebbles = list(map(int, input().split()))
    results.append(calculate_weights(pebbles))

# 결과 출력
for result in results:
    print(result)
